pkg/instancetype/instancetype.go

```diff
@@ -791,10 +791,15 @@ func (m *InstancetypeMethods) inferDefaultsFromDataSource(dataSourceName, dataSo
 	return "", "", fmt.Errorf("unable to infer defaults from DataSource that doesn't provide DataVolumeSourcePVC")
 }
 
-func (m *InstancetypeMethods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1beta1.DataVolumeSourceRef, defaultNameLabel, defaultKindLabel string) (string, string, error) {
+func (m *InstancetypeMethods) inferDefaultsFromDataVolumeSourceRef(sourceRef *v1beta1.DataVolumeSourceRef, defaultNameLabel, defaultKindLabel, vmNameSpace string) (string, string, error) {
 	switch sourceRef.Kind {
 	case "DataSource":
-		return m.inferDefaultsFromDataSource(sourceRef.Name, *sourceRef.Namespace, defaultNameLabel, defaultKindLabel)
+		// The namespace can be left blank here with the assumption that the DataSource is in the same namespace as the VM
```

Comment:
I would suggest setting the namespace just to empty string (`""`) and let CDI figure out the rest. Avoids carrying the `vmNameSpace` through several function calls.